iaso/api/instances.py

```diff
@@ -598,6 +620,68 @@ def bulkdelete(self, request):
             status=201,
         )
 
+    @action(
+        detail=False,
+        methods=["GET"],
+        permission_classes=[permissions.IsAuthenticated, HasInstanceBulkPermission, HasCreateOrgUnitPermission],
+    )
+    def check_bulk_gps_push(self, request):
+        # first, let's parse all parameters received from the URL
+        select_all, selected_ids, unselected_ids = self._parse_check_bulk_gps_push_parameters(request.GET)
+
+        # then, let's make sure that each ID actually exists and that the user has access to it
+        instances_query = self.get_queryset()
+        for selected_id in selected_ids:
+            get_object_or_404(instances_query, pk=selected_id)
+        for unselected_id in unselected_ids:
+            get_object_or_404(instances_query, pk=unselected_id)
```

Comment:
This will trigger one SQL query per iteration of each loop. And this can increase rapidly depending on the number of `selected_id` and `unselected_id`.

I think we can do something more performant with only two queries:

```python
if instances_query.filter(pk__in= selected_ids).count() != len(selected_id):
    raise Http404
if instances_query.filter(pk__in=unselected_id).count() != len(unselected_id):
    raise Http404
```